### ISSUE TITLE
Edit Textures of Blocks and their Models; Biome Air, Water, Grass, etc. Color - Procedure Blocks

### ISSUE THREAD

**Issue by ErinRoseWebs** [Team-MCToolkit/MCToolkit#34]
This would basically allow you to specify a block face and/or model texture (integer), and change it's effects, such as color changing or changing the texture all together. This would allow for furnaces with one block and no info transfer instead of two and complicated procedures. Yes, this would also include image effects. 
![mctoolkit suggestions 1](https://user-images.githubusercontent.com/67371766/98454236-41bd3d80-2130-11eb-917e-7099c488ae29.png)
I can't think of any other image effects right off the top of my head, but I gave two examples in the above image.

Also, since I think I saw someone wanting to set the color of water in a model to the biome's water color, here's a suggestion for getting the color.
![mctoolkit suggestions 2](https://user-images.githubusercontent.com/67371766/98454312-446c6280-2131-11eb-90ff-b4b7c16b460b.png)

If you think there's anything else I could add, or if there's anything you don't think can be included, let me know!

**Comment by Max094Reikeb** [comment]
Yeah, why such a complicated name for your feature request while you could just have say "Custom blockstates"?
It's basically that.

**Comment by ErinRoseWebs** [comment]
I was just specifying two things. I can simplify it more if ya want.

**Comment by Defeatomizer** [comment]
I have some ideas related to blockstates support, but it is quite difficult to implement (visual properties have to be redone from scratch), especially in case users want to define block properties basing on current state of this block.
As about colors, I think it would be better if there were two external plugins: one adds color variables support (define variable, get color from_number/as_srtring, get red/green/blue component from color and stuff like that), and another one adds suggested procedure blocks themselves.
